lib/src/sql/special.rs

```diff
@@ -95,54 +82,43 @@ pub fn check_group_by_fields<'a>(
 	// Check to see if a GROUP BY clause has been defined
 	if let Some(groups) = groups {
 		// Loop over each of the expressions in the GROUP BY clause
-		'outer: for group in groups.iter() {
-			// Loop over each of the expressions in the SELECT clause
-			for field in fields.iter() {
-				// Check to see whether the expression is in the SELECT clause
-				match field {
-					// This field is aliased, so check the alias name
-					Field::Alias(_, i) if i.as_ref() == group.as_ref() => continue 'outer,
-					// This field is not aliased, so check the field value
-					Field::Alone(v) => {
-						match v {
-							// If the expression in the SELECT clause is a field, check if it exists in the GROUP BY clause
-							Value::Idiom(i) if i.as_ref() == group.as_ref() => continue 'outer,
-							// Otherwise check if the expression itself exists in the GROUP BY clause
-							v if v.to_idiom().as_ref() == group.as_ref() => continue 'outer,
```

Comment:
Something I noticed (but didn't change) is that this file diverges from the rest by not calling `to_idiom` on `Value::Idiom`'s. This avoids some memory allocations but also means there is no call to `Idiom::simplify`. Not sure if this matters.